packages/jira-adapter/src/weak_references/field_configuration_items.ts

```diff
@@ -13,68 +13,77 @@
+import { FIELD_TYPE_NAME } from '../filters/fields/constants'
 
 const { awu } = collections.asynciterable
+const { pickAsync } = promises.object
 
 const log = logger(module)
 
+export type FieldItem = {
+  description?: string
+  isHidden?: boolean
+  isRequired?: boolean
+  renderer?: string
+}
+
 type FieldConfigurationItems = {
-  [key: string]: {
-    id: unknown
-    description: string
-    isHidden: boolean
-    isRequired: boolean
-    render: string
-  }
+  [key: string]: FieldItem
 }
 
 const FIELD_CONFIGURATION_ITEM_SCHEME = Joi.object({
-  id: Joi.required(),
   description: Joi.optional(),
-  isHidden: Joi.boolean().required(),
-  isRequired: Joi.boolean().required(),
+  isHidden: Joi.boolean().optional(),
+  isRequired: Joi.boolean().optional(),
   renderer: Joi.optional(),
 })
 
 const FIELD_CONFIGURATION_ITEMS_SCHEME = Joi.object().pattern(/.*/, FIELD_CONFIGURATION_ITEM_SCHEME)
 
-const isFieldConfigurationItems = createSchemeGuard<FieldConfigurationItems>(FIELD_CONFIGURATION_ITEMS_SCHEME, 'Received an invalid field configuration items value')
+export const isFieldConfigurationItems = createSchemeGuard<FieldConfigurationItems>(FIELD_CONFIGURATION_ITEMS_SCHEME)
 
-const getFieldReferences = async (
+const getFieldReferences = (
   instance: InstanceElement,
-): Promise<ReferenceInfo[]> => {
+): ReferenceInfo[] => {
   const fieldConfigurationItems = instance.value.fields
   if (fieldConfigurationItems === undefined || !isFieldConfigurationItems(fieldConfigurationItems)) {
     log.warn(`fields value is corrupted in instance ${instance.elemID.getFullName()}, hence not calculating fields weak references`)
     return []
   }
-  return awu(Object.entries(fieldConfigurationItems))
-    .map(async ([id, config]) => (
-      isReferenceExpression(config.id)
-        ? { source: instance.elemID.createNestedID('fields', id, 'id'),
-          target: config.id.elemID,
-          type: 'weak' as const }
-        : undefined))
+  return Object.keys(fieldConfigurationItems)
+    .map(fieldName => {
+      const elemId = new ElemID(JIRA, FIELD_TYPE_NAME, 'instance', fieldName)
+      return {
+        source: instance.elemID.createNestedID('fields', fieldName),
+        target: elemId,
+        type: 'weak' as const,
+      }
+    })
     .filter(values.isDefined)
-    .toArray()
 }
 
 /**
  * Marks each field reference in field configuration as a weak reference.
  */
-const getFieldConfigurationItemsReferences: GetCustomReferencesFunc = async elements => log.time(() => awu(elements)
+const getFieldConfigurationItemsReferences: GetCustomReferencesFunc = async elements => log.time(() => elements
   .filter(isInstanceElement)
   .filter(instance => instance.elemID.typeName === FIELD_CONFIGURATION_TYPE_NAME)
-  .flatMap(getFieldReferences)
-  .toArray(), 'getFieldConfigurationItemsReferences')
+  .flatMap(getFieldReferences), 'getFieldConfigurationItemsReferences')
+
+const fieldExists = async (
```

Comment:
another `async` with no `await`
```suggestion
const fieldExists = (
```